sensors/test_deps.go

```diff
@@ -137,3 +151,58 @@ func getFinishedReplayLidar() *inject.Camera {
 	}
 	return cam
 }
+
+func getGoodIMU() *inject.MovementSensor {
+	imu := &inject.MovementSensor{}
+	vec := r3.NewPreciseVector(1, 1, 1).Vector()
+	imu.LinearAccelerationFunc = func(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+		return vec, nil
+	}
+	imu.AngularVelocityFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+		return spatialmath.PointAngVel(vec, vec), nil
+	}
+
+	imu.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{
+			AngularVelocitySupported:    true,
+			LinearAccelerationSupported: true,
+		}, nil
+	}
+	return imu
+}
+
+func getInvalidIMU() *inject.MovementSensor {
+	imu := &inject.MovementSensor{}
+	imu.LinearAccelerationFunc = func(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+		return r3.Vector{}, errors.New("invalid sensor")
+	}
+	imu.AngularVelocityFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+		return spatialmath.AngularVelocity{}, errors.New("invalid sensor")
+	}
+	imu.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{
+			AngularVelocitySupported:    true,
+			LinearAccelerationSupported: true,
+		}, nil
+	}
+	return imu
+}
+
+func getBadIMU() *inject.MovementSensor {
+	imu := &inject.MovementSensor{}
+	vec := r3.NewPreciseVector(1, 1, 1).Vector()
+	imu.LinearAccelerationFunc = func(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+		return vec, nil
+	}
+	imu.AngularVelocityFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+		return spatialmath.PointAngVel(vec, vec), nil
+	}
+
+	imu.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{
+			AngularVelocitySupported:    false,
+			LinearAccelerationSupported: true,
```

Comment:
[opt] I would do false here as well, until we need to split off an injected imu that only supports one or the other.